docs: add explanation for use of `fetchKey` (#1750)
nuxt/nuxtjs.org: content/en/docs/3.features/4.data-fetching.md

`fetchOnServer`: `Boolean` or `Function` (default: `true`), call `fetch()` when server-rendering the page

> `fetchKey`: `String` or `Function` (defaults to the component scope ID or component name), a key (or a function that produces a unique key) that identifies the result of this component's fetch (available on Nuxt 2.15+). It is used to map SSR-data to client-data in the hydration process, where SSR-data is the component's data after calling `fetch()` during server-side rendering, and client-data is the component's data at client side. [More information available in original PR](https://github.com/nuxt/nuxt.js/pull/8466).

Reviewer: How about this?

```suggestion
`fetchKey`: `String` or `Function` (defaults to the component scope ID or component name), a key (or a function that produces a unique key) that identifies the result of this component's fetch (available on Nuxt 2.15+). This key is used to match the result of the `fetch()` during server-side rendering to the client-side component data in the hydration process. [More information available in original PR](https://github.com/nuxt/nuxt.js/pull/8466).
```
